### PR TITLE
update txt metadata

### PR DESCRIPTION
**Pull request by andre-urbani**
### What

Update txt download for BYO 
Trello - https://trello.com/c/JkyoYKdQ/6077-update-txt-download-for-byo

### How to review

Confirm changes match the requested changes in Trello ticket

### Who can review

Anyone

### CODE REVIEW

File: text/text.go [hunk 1]
```diff
@@ -4,123 +4,174 @@ import (
...
+	"time"
 
 	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
 )
 
-func NewMetadata(m *dataset.Metadata) []byte {
-	var b bytes.Buffer
+func NewMetadata(m *dataset.Metadata, isCustom bool, filterOutputID, downloadServiceURL string) []byte {
+	if isCustom {
+		var b bytes.Buffer
```
Comment (nshumoogum) [inline text/text.go:14]: Could be worth splitting custom dataset metadata and non custom metadata separated into two separate functions? That way easier to test logic

File: text/text.go [hunk 2]
```diff
@@ -4,123 +4,174 @@ import (
...
 	"fmt"
 	"reflect"
+	"time"
 
 	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
 )
 
-func NewMetadata(m *dataset.Metadata) []byte {
-	var b bytes.Buffer
+func NewMetadata(m *dataset.Metadata, isCustom bool, filterOutputID, downloadServiceURL string) []byte {
```
Comment (nshumoogum) [inline text/text.go:12]: Function could do with some unit tests

File: text/text.go [hunk 3]
```diff
@@ -4,123 +4,174 @@ import (
...
 
-	b.WriteString("Usage Notes:\n")
-	if m.DatasetDetails.UsageNotes != nil {
-		for _, n := range *m.DatasetDetails.UsageNotes {
-			b.WriteString(fmt.Sprintf("Title: %s\nNote: %s\n\n", n.Title, n.Note))
+		b.WriteString(fmt.Sprintf("Title: %s\n", titleDims))
+		b.WriteString(fmt.Sprintf("Issued: %s\n", issuedDate))
+		b.WriteString(fmt.Sprintf("Language: %s\n", "English"))
+		b.WriteString("Distribution:\n")
+		for k, v := range m.Downloads {
```
Comment (nshumoogum) [inline text/text.go:27]: Is it not possible to do the same for loop for non custom dataset? Like so:

```
		for k, v := range m.Downloads {
			b.WriteString(fmt.Sprintf("\tExtension: %s\n", k))
			b.WriteString(fmt.Sprintf("\tSize: %s\n", v.Size))
			b.WriteString(fmt.Sprintf("\tURL: %s\n\n", v.URL))
		}
```

If yes, then can move this to a separate function and can be used by both sections

**Comment by andre-urbani** [comment]
I'm going to merge to develop for now as this ideally needs to be ready for "go live" tomorrow. I will work on the requested changes afterwards

File: text/text.go (hunk 3 above)
Comment (andre-urbani) [reply text/text.go:27]: I don't think this is possible as v.URL does not contain the correct url, therefore they need to be processed separately depending on the file type

**Pull request merged**